addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1672,6 +1682,24 @@ options.registry.anchorName = options.Class.extend({
         }
         this.$target.trigger('content_changed');
     },
+    /**
+     * Returns anchor text.
+     *
+     * @private
+     */
+     _getAnchorLink: function (previewMode, value, $opt) {
+         if (!this.$target[0].id) {
+             const $titles = this.$target.find('h1, h2, h3, h4, h5, h6');
+             const title = $titles.length > 0 ? $titles[0].innerText : this.data.snippetName;
+             const anchorName = encodeURIComponent(title.trim().replace(/\s/g, '-'));
```

Comment:
This transformation has to be put in a private function as it is used twice (in the dialog and here)... and in fact in the dialog you had the good idea to use `/\s+/g` but not here :wink: